index.js

```diff
@@ -25,61 +25,61 @@ instance.prototype.updateConfig = function(config) {
 	self.init_tcp();
 };
 
+instance.prototype.init = function() {
+	var self = this;
+
+	debug = self.debug;
+	log = self.log;
+
+	self.init_tcp();
+};
+
 instance.prototype.incomingData = function(data) {
 	var self = this;
 	debug(data);
 
 	// Match part of the copyright response from unit when a connection is made.
-	// Send Info request which should reply with Matrix setup, eg: "V8X4 A8X4"
 	if (self.login === false && data.match("Extron Electronics")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write("I"+ "\n");
+		self.socket.write("I"+ "\n"); // Matrix information request
 	}
 
 	if (self.login === false && data.match("Password:")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write(""+ "\n");
+		self.socket.write("\r" +self.config.password+ "\r"); // Enter Password Set
 	}
 
-	// Match first letter of expected response from unit.
-	else if (self.login === false && data.match(/V|60/)) {
+	// Match login sucess response from unit.
+	else if (self.login === false && data.match("Login")) {
 		self.login = true;
 		self.status(self.STATUS_OK);
 		debug("logged in");
 	}
-	else if (self.login === false && data.match('login incorrect')) {
-		self.log('error', "incorrect username/password (expected no password)");
-		self.status(self.STATUS_ERROR, 'Incorrect user/pass');
+	// Match expected response from unit.
+	else if (self.login === false && data.match(/V|60-/)) {
+		self.login = true;
+		self.status(self.STATUS_OK);
+		debug("logged in");
 	}
 	// Heatbeat to keep connection alive
+	function heartbeat() {
+		self.login = false;
+		self.status(self.STATUS_WARNING,'Checking Connection');
+		self.socket.write("N"+ "\n"); // should respond with Switcher part number eg: "60-882-01" = DXP 88 HDMI
+		debug("Checking Connection");
+		}
```

Comment:
Not to be nitpicky, but the right curly bracket here is indented wrong. Also, you could change "N"+"\n", to simply "N\n" ;)

Reply:
Be nitpicky! I really appreciate it :) appeals to my OCD.